examples/step-60/step-60.cc

```diff
@@ -0,0 +1,1077 @@
+  //
+  // A novelty w.r.t. other tutorial programs is the heavy use of
+  // std::unique_ptr. These behave like classical pointers, with the advantage
+  // of doing automatic house-keeping: the contained object is automatically
+  // destroyed as soon as the unique_ptr goes out of scope, even if it is inside
+  // a container or there's an exception. Moreover it does not allow for
+  // duplicate pointers, which prevents ownership problems. We do this, because
+  // we want to be able to i) construct the problem, ii) read the parameters,
+  // and iii) initialize all objects according to what is specified in a
+  // parameter file.
+  //
+  // We construct the parameters of our problem in the internal class
+  // DistributedLagrangeProblemParameters, derived from ParameterAcceptor. The
+  // DistributedLagrangeProblem class takes a const reference to a
+  // DistributedLagrangeProblemParameters object, so that it is not possible to
+  // modify the parameters from within the DistributedLagrangeProblem class
+  // itself.
+  //
+  // We could have initialized the parameters first, and then pass the
+  // parameters to the DistributedLagrangeProblem assuming all entries are set to
+  // the desired values, but this has two disadvantages:
+  //
+  // - we should not make assumptions on how the user initializes a class that
+  // is not under our direct control. If the user fails to initialize the
+  // class, we should notice and throw an exception;
+  //
+  // - not all objects that need to read parameters from a parameter file may
+  // be available when we construct the DistributedLagrangeProblemParameters;
+  // this is often the case for complex programs, with multiple physics, or
+  // where we reuse existing code in some external classes. We simulate this by
+  // keeping some "complex" objects, like ParsedFunction objects, inside the
+  // DistributedLagrangeProblem instead of inside the
+  // DistributedLagrangeProblemParameters.
+  //
+  // Here we assume that upon construction, the classes that build up our
+  // problem are not usable yet. Parsing the parameter file is what ensures we have
+  // all ingredients to build up our classes, and we design them so that if parsing
+  // fails, or is not executed, the run is aborted.
+
+  template <int dim, int spacedim=dim>
+  class DistributedLagrangeProblem
+  {
+  public:
+
+    // The DistributedLagrangeProblemParameters is derived from
+    // ParameterAcceptor. This allows us to use the
+    // ParameterAcceptor::add_parameter methods in its constructor.
+    //
+    // The members of this function are all non-const, but the
+    // DistributedLagrangeProblem class takes a const reference to a
+    // DistributedLagrangeProblemParameters object: this ensures that
+    // parameters are not modified from within the DistributedLagrangeProblem
+    // class.
+
+    class DistributedLagrangeProblemParameters : public ParameterAcceptor
```

Comment:
see the comment above

Reply:
also, maybe remove the empty line so that the comment immediately attached to the class